src/components/Popups/PopupHighlight.tsx

```diff
@@ -60,19 +60,33 @@ export default function PopupHighlight({ onCancel = noop, onClick = noop, shape
         }),
     };
 
-    const handleClick = (event: React.MouseEvent): void => {
-        onClick(event);
+    const handleKeydownOutside = (event: Event): void => {
+        if (!(event instanceof KeyboardEvent) || event.key !== 'Enter' || event.ctrlKey || event.metaKey) {
+            return;
+        }
+
+        event.preventDefault(); // Consume any enter keydown event if the promoter is open
+        event.stopPropagation(); // Prevent the enter keydown event from adding a new line to the textarea
+
+        onSubmit();
     };
 
+    useOutsideEvent('keydown', buttonRef, handleKeydownOutside);
     useOutsideEvent('mousedown', buttonRef, onCancel);
 
     return (
-        <PopupBase className="ba-PopupHighlight" options={options} reference={reference}>
+        <PopupBase
+            aria-live="polite"
```

Comment:
It seems to work well. I found that omitting `aria-live` made it so that VoiceOver doesn't "see" the popup when it's added to the DOM. I think `role="dialog"` should be okay, from the docs linked:

>The dialog role is used to mark up an HTML based application dialog or window that separates content or UI from the rest of the web application or page. Dialogs are generally placed on top of the rest of the page content using an overlay. Dialogs can be either non-modal (it's still possible to interact with content outside of the dialog) or modal (only the content in the dialog can be interacted with).

Reply:
Should we have some `aria-label` for the dialog to instruct about pressing 'enter' to add the highlight? I'm thinking since it's not a standard a11y pattern, they might not know what to do even if the screen reader reads out the promoter

Reply:
I don't think so. It's a button, so it should be the same interaction as any other interactive element.